components/brave_wallet_ui/common/async/hardware.ts

```diff
@@ -93,11 +100,11 @@ export async function signLedgerEthereumTransaction (
     return { success: false, error: getLocale('braveWalletApproveTransactionError') }
   }
   const data = await apiProxy.txService.getTransactionMessageToSign(coin, txInfo.id)
-  if (!data || !data.message) {
+  if (!data || !data.message || !data.message.messageStr) {
     return { success: false, error: getLocale('braveWalletNoMessageToSignError') }
   }
 
-  const signed = await deviceKeyring.signTransaction(path, data.message.replace('0x', ''))
+  const signed = await deviceKeyring.signTransaction(path, data.message.messageStr?.replace('0x', ''))
```

Comment:
I change `data.message` to `data.message.messageStr` because of the change in the [GetTransactionMessageToSign](https://github.com/brave/brave-core/pull/13734/files#diff-9f2cf11d57c8efee5461b92cdb38fa327b70c25fd8d009f7692049e8e88944a8R908) mojo API, which now can either return a string, or uint8 array (bytes).  This change is made because the Ethereum ledger expects that the messages we send to be strings, whereas the the Solana ledger library expects that the messages are bytes.